File: gherkin/java/src/main/java/gherkin/pickles/PickleTable.java
package gherkin.pickles;

import java.util.List;

import static java.util.Collections.unmodifiableList;

public class PickleTable implements Argument {

    private final List<PickleRow> rows;

    public PickleTable(List<PickleRow> rows) {
        this.rows = unmodifiableList(rows);
    }

    public List<PickleRow> getRows() {
        return rows;
    }

    @Override
    public PickleLocation getLocation() {
        return rows.get(0).getCells().get(0).getLocation();
    }
}

File: gherkin/dotnet/Gherkin/Pickles/PickleTable.cs
using System.Collections.Generic;
using System.Runtime.Serialization;

namespace Gherkin.Pickles
{
    public class PickleTable
    {
        [DataMember(Name = "rows")]
        public IEnumerable<PickleRow> Rows { get; private set; }

        public PickleTable()
        {
            
        }
        
        public PickleTable(IEnumerable<PickleRow> rows)
        {
            Rows = rows;
        }
    }
}
